### PR TITLE
storage retry operation/7907

### PR DESCRIPTION
**Pull request by HemangChothani**
issue [7907]
pmtask [4419]

### CODE REVIEW

File: storage/google/cloud/storage/bucket.py [hunk 1]
```diff
@@ -545,7 +547,7 @@ def exists(self, client=None):
         except NotFound:
             return False
 
-    def create(self, client=None, project=None, location=None):
+    def create(self, retry=DEFAULT_RETRY, client=None, project=None, location=None):
```
Comment (mf2199) [inline storage/google/cloud/storage/bucket.py]: This way is better, as it minimizes changes to the interface:
`def create(self, client=None, project=None, location=None, retry=DEFAULT_RETRY):`

File: storage/google/cloud/storage/client.py [hunk 2]
```diff
@@ -476,6 +490,13 @@ def list_buckets(
             max_results=max_results,
             extra_params=extra_params,
         )
+        return api_response
+
+    def _call_api(self, retry, **kwargs):
+        call = functools.partial(page_iterator.HTTPIterator, **kwargs)
+        if retry:
+            call = retry(call)
+        return call()
```
Comment (mf2199) [inline storage/google/cloud/storage/client.py]: This function is similar to the one found in blob.py and bucket.py. Is it possible to combine them in a single method in the base class, _helpers/_PropertyMixin? Something of this kind:
```
    def _call_api(self, fcn_call, retry, **kwargs):
        return retry(fcn_call) if retry else fcn_call()
```

File: storage/tests/unit/test_bucket.py [hunk 3]
```diff
@@ -14,8 +14,37 @@
...
+    response.request = requests.Request()
+    return response
+
+
+def _make_requests_session(responses):
+    session = mock.create_autospec(requests.Session, instance=True)
+    session.request.side_effect = responses
+    return session
+
+
```
Comment (mf2199) [inline storage/tests/unit/test_bucket.py:39]: Keep single empty line between methods, unless entire file uses double-lines.

Comment (HemangChothani) [reply storage/tests/unit/test_bucket.py:39]: That is PEP8 Style guide for python code, so in that if methods are not in class or in main need to keep two blank line between class or methods but if methods in the class than use single blank line between them, if you are using Pycharm IDEE for python than you found instruction about PEP8 when you don't follow that guide line

File: storage/google/cloud/storage/bucket.py (hunk 1 above)
Comment (HemangChothani) [reply storage/google/cloud/storage/bucket.py]: yes , i found that , may be in hurry i did that mistake , I will change that

File: storage/google/cloud/storage/client.py (hunk 2 above)
Comment (HemangChothani) [reply storage/google/cloud/storage/client.py]: Yes,Actually i don't  find common.py file so don't know which file is using as a common file ,Now i move that function to _helpers.py file and remove from client,blob,bucket files

**Pull request closed without merging**